### PR TITLE
Updating Bettercamera (also fixes cannons)

### LINKED ISSUE #116: Analog camera reverts to vanilla camera hitting zoom out in cannon
Version: Refresh_8.6_Windows
Stage: Jolly Roger Bay
Running windowed, default resolution when false is set in config

Steps to reproduce:
Enter Jolly Roger Bay, Choose first star (Plunder in the Sunken Ship)
Enter the cannon
After you're able to aim press R (Or whatever key you have mapped to zoom out camera)
The camera icon will switch to Mario's head
Launch the cannon

The camera should be reverted back to vanilla controls now.

My Config:
`fullscreen false
key_a 38
key_b 51
key_start 57
key_l 52
key_r 54
key_z 37
key_cup 328
key_cdown 336
key_cleft 331
key_cright 333
key_stickup 17
key_stickdown 31
key_stickleft 30
key_stickright 32
joy_a 0
joy_b 2
joy_start 6
joy_l 7
joy_r 10
joy_z 9
mouse_a 3
mouse_b 1
mouse_l 4
mouse_r 5
mouse_z 2
bettercam_enable true
bettercam_mouse_look false
bettercam_invertx false
bettercam_inverty false
bettercam_xsens 50
bettercam_ysens 35
bettercam_aggression 50
bettercam_pan_level 0
`

### PR DESCRIPTION
**Pull request by coltongit**
Updates bettercamera to the latest version. Also adds a cannon fix.
Fixes #116.

### CODE REVIEW

File: src/game/camera.c [hunk 1]
```diff
@@ -3080,6 +3080,12 @@ void update_camera(struct Camera *c) {
     sCButtonsPressed = find_c_buttons_pressed(sCButtonsPressed, gPlayer1Controller->buttonPressed,gPlayer1Controller->buttonDown);
 #ifdef BETTERCAMERA
     }
+
+    if (gMarioState->action == ACT_SHOT_FROM_CANNON && newcam_active)
+    {
+        gMarioState->area->camera->mode = CAM_MODE_NEWCAM;
+        gLakituState.mode = CAM_MODE_NEWCAM;
```
Comment (porcino) [inline src/game/camera.c:3087]: you meant CAMERA_MODE_NEWCAM?

File: src/game/bettercamera.inc.h [hunk 2]
```diff
@@ -264,27 +272,33 @@ static void newcam_rotate_button(void)
     if (newcam_modeflags & NC_FLAG_XTURN)
     {
         if ((gPlayer1Controller->buttonDown & L_CBUTTONS) && newcam_analogue == 0)
-            newcam_yaw_acc = newcam_adjust_value(newcam_yaw_acc,accel);
+            newcam_yaw_acc = newcam_adjust_value(newcam_yaw_acc,accel, 100);
         else if ((gPlayer1Controller->buttonDown & R_CBUTTONS) && newcam_analogue == 0)
-            newcam_yaw_acc = newcam_adjust_value(newcam_yaw_acc,-accel);
+            newcam_yaw_acc = newcam_adjust_value(newcam_yaw_acc,-accel, -100);
```
Comment (porcino) [inline src/game/bettercamera.inc.h:277]: I got `error: too many arguments to function 'newcam_adjust_value'` here

File: src/game/bettercamera.inc.h [hunk 3]
```diff
@@ -344,18 +360,23 @@ static void newcam_rotate_button(void)
                 }
             }
             else
-                newcam_yaw_acc = newcam_adjust_value(newcam_yaw_acc,(-gPlayer2Controller->stickX/4));
-        }
+            {
+                newcam_yaw_acc = newcam_adjust_value(newcam_yaw_acc,(-gPlayer2Controller->stickX/8, intendedXMag));
+            }
```
Comment (porcino) [inline src/game/bettercamera.inc.h:365]: `error: expected '}' before 'else'`

File: src/game/bettercamera.inc.h [hunk 4]
```diff
@@ -189,12 +189,18 @@ void newcam_diagnostics(void)
...
+        var += val;
+        if (var > max)
+            var = max;
+    }
+    else if (val < 0)
+    {
+        var += val;
+        if (var < max)
+            var = max;
+    }
```
Comment (porcino) [inline src/game/bettercamera.inc.h:203]: `error: 'max' undeclared (first use in this function); did you mean 'fmax'?`

File: src/game/camera.c (hunk 1 above)
Comment (coltongit) [reply src/game/camera.c]: Fixed.

File: src/game/bettercamera.inc.h (hunk 2 above)
Comment (coltongit) [reply src/game/bettercamera.inc.h:277]: Fixed.

File: src/game/bettercamera.inc.h (hunk 4 above)
Comment (coltongit) [reply src/game/bettercamera.inc.h:203]: Fixed.

File: src/game/bettercamera.inc.h (hunk 3 above)
Comment (coltongit) [reply src/game/bettercamera.inc.h:365]: Try now and see if this commit fixes this

**Comment by porcino** [comment]
almost: 
Untitled

File: src/game/bettercamera.inc.h (hunk 3 above)
Comment (coltongit) [reply src/game/bettercamera.inc.h:365]: Alright. Resolved for good.

**Comment by coltongit** [comment]
@porcino try now

**Pull request merged**